parsing/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -967,7 +979,158 @@ SqlCreate SqlCreateTable() :
     {
         return new SqlCreateTableDialect1(s.end(this), createSpecifier, setType,
          volatility, ifNotExists, id, tableAttributes, columnList, query,
-         withData, primaryIndex, indices, onCommitType);
+         withData, primaryIndex, indices, partition, onCommitType);
+    }
+}
+
+SqlTablePartition CreateTablePartitionBy() :
+{
+    final SqlNodeList partitions = new SqlNodeList(getPos());
+    SqlNode e;
+}
+{
+    (
+        e = PartitionExpression()
+        { partitions.add(e); }
+    |
+        <LPAREN>
+        e = PartitionExpression()
+        { partitions.add(e); }
+        (
+            <COMMA>
+            e = PartitionExpression()
+            { partitions.add(e); }
+        )*
+        <RPAREN>
+    )
+    { return new SqlTablePartition(getPos(), partitions); }
+}
+
+SqlNode  PartitionExpression() :
+{
+    final SqlNode e;
+    int constant = 0;
+}
+{
+    (
+        e = RangeN()
+    |
+        e = CaseN()
+    |
+        e = SqlExtractFromDateTime()
+    |
+        e = PartitionByColumnOption()
+    )
+    [
+        <ADD> { constant = UnsignedIntLiteral(); }
+    ]
+    { return new SqlTablePartitionExpression(getPos(), e, constant); }
+}
+
+SqlNode PartitionByColumnOption() :
+{
+    SqlNode e;
+    boolean containAllButSpecifier = false;
+    final SqlNodeList columnList = new SqlNodeList(getPos());
+}
+{
+    (
+        <COLUMN> <ALL> <BUT> <LPAREN>
+        e = PartitionColumnItem()
+        { columnList.add(e); }
+        (
+            <COMMA>
+            e = PartitionColumnItem()
+            { columnList.add(e); }
+        )*
+        <RPAREN>
+        { containAllButSpecifier = true; }
+    |
+        <COLUMN> <LPAREN>
+        e = PartitionColumnItem()
+        { columnList.add(e); }
+        (
+            <COMMA>
+            e = PartitionColumnItem()
+            { columnList.add(e); }
+        )*
+        <RPAREN>
+    |
+        <COLUMN>
+        {
+            return new SqlTablePartitionByColumn(getPos(), columnList, false);
+        }
+    |
+        <COLUMN>
+        (
+            e = SimpleIdentifier()
+            {columnList.add(e);}
```

Comment:
nit: `{ columnList.add(e); }`

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/219/commits/6f8d1b7698fb19b088565e8b1e84d5a144a6c10c